src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.cc

```diff
@@ -16,34 +16,327 @@
  *
  */
 
+#include <grpc/support/port_platform.h>
+
 #include "src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.h"
 
-namespace grpc {
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+#include <grpc/support/sync.h>
+
+#include "src/core/lib/slice/slice_internal.h"
+#include "src/core/lib/surface/channel.h"
+#include "src/core/tsi/s2a/handshaker/s2a_handshaker_client.h"
+#include "src/core/tsi/s2a/s2a_constants.h"
+
+namespace grpc_core {
 namespace experimental {
 
+/** The main struct for the S2A TSI handshaker. **/
+struct s2a_tsi_handshaker {
+  tsi_handshaker base;
+  grpc_slice target_name;
+  bool is_client;
+  bool has_sent_start_message;
+  bool has_created_handshaker_client;
+  char* handshaker_service_url;
+  grpc_pollset_set* interested_parties;
+  grpc_s2a_credentials_options* options;
+  s2a_handshaker_client_vtable* client_vtable_for_testing;
+  grpc_channel* channel;
+  gpr_mu mu;
```

Comment:
How about adding some comments about `mu` here? similar to what we did for ALTS?

Reply:
Added a comment analogous to the ALTS one.